presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```
@@ -0,0 +1,458 @@
+
+            return filter;
+        }
+
+        private static ImmutableMap<Set<Integer>, DeleteSetInfo> collectDeleteInformation(Table icebergTable,
+                IcebergTableHandle icebergTableHandle,
+                long snapshotId)
+        {
+            HashMap<Set<Integer>, DeleteSetInfo> deleteInformations = new HashMap<>();
+            try (CloseableIterator<DeleteFile> files =
+                    IcebergUtil.getDeleteFiles(icebergTable, snapshotId, icebergTableHandle.getPredicate(), Optional.empty()).iterator()) {
+                while (files.hasNext()) {
+                    DeleteFile delete = files.next();
+                    if (delete.content() == FileContent.EQUALITY_DELETES) {
+                        Map<Integer, PartitionFieldInfo> partitionFields = new HashMap<>();
+                        PartitionSpec partitionSpec = icebergTable.specs().get(delete.specId());
+                        Types.StructType partitionType = partitionSpec.partitionType();
+                        // PartitionField ids are unique across the entire table in v2. We can assume we are in v2 since v1 doesn't have equality deletes
+                        partitionSpec.fields().forEach(f -> partitionFields.put(f.fieldId(), new PartitionFieldInfo(partitionType.field(f.fieldId()), f)));
+                        List<Integer> partitionFieldIds = partitionSpec.fields().stream().map(PartitionField::fieldId).collect(Collectors.toList());
+                        HashSet<Integer> result = new HashSet<>();
+                        result.addAll(delete.equalityFieldIds());
+                        result.addAll(partitionFieldIds);
+                        deleteInformations.put(ImmutableSet.copyOf(result), new DeleteSetInfo(partitionFields, delete.equalityFieldIds()));
+                    }
+                }
+            }
+            catch (IOException e) {
+                return ImmutableMap.of();
```

I think this should fail loudly.  Consider an Iceberg exception.